src/services.js

```diff
@@ -632,6 +632,26 @@ export class Services {
     ));
   }
 
+  /**
+   * @param {!Element|!./service/ampdoc-impl.AmpDoc} elementOrAmpDoc
+   * @return {!Promise<!./service/storage-impl.Storage>}
+   */
+  static storageForTopLevelDoc(elementOrAmpDoc) {
+    const thisAmpdoc = Services.ampdoc(elementOrAmpDoc);
+    const ampdocService = Services.ampdocServiceFor(thisAmpdoc.win);
+    const topAmpdoc = ampdocService.isSingleDoc()
+      ? ampdocService.getSingleDoc()
+      : null;
+    // We need to verify that ampdocs are on the same origin, therefore
+    // we compare the windows of both.
+    const ampdoc =
+      topAmpdoc && topAmpdoc.win == thisAmpdoc.win ? topAmpdoc : thisAmpdoc;
+    return /** @type {!Promise<!./service/storage-impl.Storage>} */ (getServicePromiseForDoc(
+      ampdoc,
+      'storage'
+    ));
+  }
```

Comment:
Can we add a test for this new functionality in `test-storage`?

Reply:
`test-service.js` is for `service.js`. Unfortunately, there's are no unit tests for the functions in `serviceS.js` 😞

Reply:
I see, that's probably because most of the `serviceS.js` is installing element or doc services and not much meat. Up to you if you want to add test somewhere to give us more confidence.

Reply:
Added todo. Good item for fixit week.